src/java.base/share/native/libjli/java.c

```diff
@@ -534,6 +582,329 @@ invokeInstanceMainWithoutArgs(JNIEnv *env, jclass mainClass) {
     return 1;
 }
 
+#if defined(J9VM_OPT_CRAC_SUPPORT)
+/**
+ * Get the value of a command line option in the array of command line arguments.
+ * @param[in] optionName The name of the command line option to search for
+ * @param[in] argc The number of command line arguments
+ * @param[in] argv The array of command line arguments
+ * @param[out] error A pointer to an integer for the error code
+ * @return A pointer to the value of the command line option if found, NULL otherwise
+ */
+static const char *
+getCommandLineOptionValue(const char *optionName, int argc, char **argv, int *error)
+{
+    const char *value = NULL;
+    int i = 0;
+    size_t optionNameLength = strlen(optionName);
+    jboolean optionNameFound = JNI_FALSE;
+    *error = 0;
+    for (; i < argc; i++) {
```

Comment:
Even thought it is redundant here, specifying the initial value is good practice:
```c
    for (i = 0; i < argc; i++) {
```
The first value of `i` is quite clear, and you don't need to go looking elsewhere.

Reply:
How can I request clarification as it seems like the good practice is to initialize `i` at the start of the function during declaration and with the for loop in the following way:

```c
    for (i = 0; i < argc; i++) {
```

even though the first value of `i` is quite clear, and we don't need to go looking elsewhere

?

Reply:
@tajila @keithc-ca how can I know the answer to the following question: how can I request clarification as it seems like the good practice is to initialize i at the start of the function during declaration and with the for loop in the following way:

```c
    for (i = 0; i < argc; i++) {
```

even though the first value of i is quite clear, and we don't need to go looking elsewhere

?

Reply:
I wasn't suggesting that you remove the initial value from the declaration.
The optimizer will eliminate any actual redundant assignments, but better to include the initial value in each `for` loop. If a second loop were added later, it would need the initializer clause, but shouldn't be coded differently than this one.